target_chains/sui/contracts/sources/pyth_accumulator.move

```diff
@@ -0,0 +1,116 @@
+        let merkle_root_hash = deserialize::deserialize_vector(&mut msg_payload_cursor, 20);
+        cursor::take_rest<u8>(msg_payload_cursor);
+        bytes20::new(merkle_root_hash)
+    }
+
+    // Note: this parsing function is adapted from the Aptos Pyth parse_price_feed_message function
+    fun parse_price_feed_message(message_cur: &mut Cursor<u8>, clock: &Clock): PriceInfo {
+        let message_type = deserialize::deserialize_u8(message_cur);
+
+        assert!(message_type == 0, 0); // PriceFeedMessage variant
+        let price_identifier = price_identifier::from_byte_vec(deserialize::deserialize_vector(message_cur, 32));
+        let price = deserialize::deserialize_i64(message_cur);
+        let conf = deserialize::deserialize_u64(message_cur);
+        let expo = deserialize::deserialize_i32(message_cur);
+        let publish_time = deserialize::deserialize_u64(message_cur);
+        let _prev_publish_time = deserialize::deserialize_i64(message_cur);
+        let ema_price = deserialize::deserialize_i64(message_cur);
+        let ema_conf = deserialize::deserialize_u64(message_cur);
+        let price_info = price_info::new_price_info(
+            clock::timestamp_ms(clock) / 1000, // not used anywhere kept for backward compatibility
+            clock::timestamp_ms(clock) / 1000,
+            price_feed::new(
+                price_identifier,
+                pyth::price::new(price, conf, expo, publish_time),
+                pyth::price::new(ema_price, ema_conf, expo, publish_time),
+            )
+        );
+        price_info
+    }
+
+    // parse_and_verify_accumulator_updates takes as input a merkle root and cursor over the encoded update message (containing encoded
+    // leafs and merkle proofs), iterates over each leaf/proof pair and verifies it is part of the tree, and finally outputs the set of
+    // decoded and authenticated PriceInfos.
+    fun parse_and_verify_accumulator_updates(cursor: &mut Cursor<u8>, merkle_root: Bytes20, clock: &Clock): vector<PriceInfo> {
+        let update_size = deserialize::deserialize_u8(cursor);
+        let price_info_updates: vector<PriceInfo> = vector[];
+        while (update_size > 0) {
+            let message_size = deserialize::deserialize_u16(cursor);
+            let message = deserialize::deserialize_vector(cursor, (message_size as u64)); //should be safe to go from u16 to u16
+            let message_cur = cursor::new(message);
+            let price_info = parse_price_feed_message(&mut message_cur, clock);
+            cursor::take_rest(message_cur);
+
+            vector::push_back(&mut price_info_updates, price_info);
+
+            // isProofValid pops the next merkle proof from the front of cursor and checks if it proves that message is part of the
+            // merkle tree defined by merkle_root
+            assert!(merkle_tree::isProofValid(cursor, merkle_root, message), E_INVALID_PROOF);
```

Comment:
another nitpick: in a separate PR, can you rename the merkle_tree functions so they're also snake_case instead of camelCase?